fastify.js

```diff
@@ -501,6 +503,23 @@ function build (options) {
         return
       }
 
+      // It can happen that a user register a plugin with some hooks/middlewares *after*
+      // the route registration. To be sure to load also that hoooks/middlwares,
+      // we must listen for the avvio's preReady event, and update the context object accordingly.
+      app.once('preReady', () => {
```

Comment:
Why not use this on line [442](https://github.com/fastify/fastify/pull/762/files#diff-20da0b6140a4768d182af0d9b185ac28R442) to replace  `_fastify.after(...`?

Reply:
Because if we do it we will throw schema and route errors outside the `avvio`'s lifecycle.

Reply:
Oh that makes sense :)